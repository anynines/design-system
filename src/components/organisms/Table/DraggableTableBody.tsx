import React from 'react'
import { Row, TableBodyProps } from 'react-table'

import DraggableTableRow, { DraggableTableRowCategory } from './DraggableTableRow'
import { TableRow, TableAccessor, TableColumnCell, TableColumnCellColor, TableColumnIcon, TableRowColor } from './Table'

interface DraggableTableBodyProps {
  items: Row<TableRow>[]
  prepareRow: (row: Row<TableRow>) => void
  tableBodyProps: TableBodyProps
  getTableColumnColor: (type: TableAccessor | null) => TableColumnCellColor | null
  getTableColumnType: (type: TableAccessor | null) => TableColumnCell | null
  getTableColumnIconType: (type: TableAccessor | null) => TableColumnIcon | null
  disabledCategories: string[]
  color?: TableRowColor
}

const DraggableTableBody: React.FC<DraggableTableBodyProps> = (
  props
): JSX.Element => {
  const {
    items, prepareRow, tableBodyProps, getTableColumnColor, getTableColumnType,
    getTableColumnIconType, disabledCategories, color
  } = props
  const rows: JSX.Element[] = []

  const isNewCategoryFound = (index: number): boolean => {
    const currentCategory = items[index].original.category.toLowerCase()
    const previousCategory = items[index - 1].original.category.toLowerCase()
    return index - 1 >= 0 && currentCategory !== previousCategory
  }

  items.forEach((item: Row<TableRow>, index: number) => {
    prepareRow(item)

    if (index === 0 || isNewCategoryFound(index)) {
      rows.push(
        <DraggableTableRowCategory
          key={`${index.toString()}.-1`}
          index={index}
          row={items[index]}
          idx={`${index.toString()}.-1`}
        />
      )
    }

    rows.push(
      <DraggableTableRow
        key={index.toString()}
        index={index}
        row={item}
        disabled={disabledCategories.includes(item.original.category)}
        isDraggable={disabledCategories.includes(item.original.category)}
        getTableColumnColor={getTableColumnColor}
        getTableColumnType={getTableColumnType}
        getTableColumnIconType={getTableColumnIconType}
        color={color}
      />
    )
  })

  return (
    <tbody {...tableBodyProps}>
      {rows}
    </tbody>
  )
}

export default DraggableTableBody
