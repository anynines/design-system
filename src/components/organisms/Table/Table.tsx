import React from 'react'
import styled from 'styled-components'
import { useTable, useSortBy, Row } from 'react-table'

import { SortableTable, SortableTableProps } from './SortableTable'
import { DraggableTable, DraggableTableProps } from './DraggableTable'

// T Y P E S
export interface RowsDataObject {
  [key: string]: Row<TableRow>[]
}

export interface TableData {
  [key: string]: TableRow[]
}

export interface TableRow {
  page: string
  status: string
  permissions: string
  languages: string[]
  views: number
  authors: string[]
  sections: string[]
  category: string
  id?: number
  slug?: string
  title?: string
  name?: string
  published?: boolean
}

export type TableSortOption = 'draggable' | 'sortable'

export type TableAccessor = 'page' | 'status' | 'permissions' | 'languages' | 'views' | 'authors' | 'sections' | 'category'

export interface TableColumn {
  Header: TableAccessor
  accessor: TableAccessor
  sortType?: string
}

export type TableColumnCellColor = 'blue' | 'black'

export type TableColumnCell = 'sticker' | 'icons' | 'icon'

export type TableColumnIcon = 'icon'

export interface TableProps {
  tableHeaderData: TableColumn[]
  getTableColumnColor: (type: TableAccessor | null) => TableColumnCellColor | null
  getTableColumnType: (type: TableAccessor | null) => TableColumnCell | null
  getTableColumnIconType: (type: TableAccessor | null) => TableColumnIcon | null
  type: TableSortOption
  initialPages: TableRow[]
  pagesPerFolder?: number
  folderLimit?: number
}

export interface SortableTableDataProps extends TableProps {
  type: 'sortable'
}

export interface DraggableTableDataProps extends TableProps {
  type: 'draggable'
  disabledCategories: string[]
  setPages?: React.Dispatch<React.SetStateAction<TableRow[]>>
  pages?: TableRow[]
}

// C O M P O N E N T
export const Table: React.FC<DraggableTableDataProps | SortableTableDataProps> = (props) => {
  const {
    tableHeaderData,
    getTableColumnColor,
    getTableColumnIconType,
    getTableColumnType,
    type,
    initialPages,
    pagesPerFolder = 10,
    folderLimit = 5
  } = props

  const [localPages, setLocalPages] = React.useState<TableRow[]>(
    [] as TableRow[]
  )

  React.useEffect(() => {
    if (initialPages.length) setLocalPages(initialPages)
  }, [props, initialPages])

  const decideTableData = (): TableRow[] => {
    if (type !== 'draggable') return localPages

    props = props as DraggableTableDataProps
    const { pages } = props
    if (pages && pages.length > 0) return pages
    return localPages
  }

  const sortCategoryAlphabeticallyAndControlLimits = (pagesDataObject: TableData): TableRow[] => {
    let pagesDataArray: [string, TableRow[]][] = Object.entries(pagesDataObject)
      .sort((a, b) => {
        const categoryA: string = a[0].toUpperCase()
        const categoryB: string = b[0].toUpperCase()
        return categoryA.charCodeAt(0) - categoryB.charCodeAt(0)
      })

    pagesDataArray = pagesDataArray.slice(0, folderLimit)
    pagesDataArray = pagesDataArray.map((pagesArray) => {
      const slicedArray = pagesArray[1].slice(0, pagesPerFolder)
      pagesArray[1] = slicedArray
      return pagesArray
    })

    let sortedPages: TableRow[] = []
    pagesDataArray.forEach((pages) => {
      sortedPages = [...sortedPages, ...pages[1]]
    })

    return sortedPages
  }

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow
  } = useTable({ columns: tableHeaderData, data: decideTableData() }, useSortBy)

  let tableProps: SortableTableProps | DraggableTableProps

  if (type === 'draggable') {
    props = props as DraggableTableDataProps
    const { disabledCategories, pages = [], setPages } = props
    tableProps = {
      rowsData: rows,
      tableProps: getTableProps(),
      prepareRow,
      tableBodyProps: getTableBodyProps(),
      headerGroups,
      getTableColumnColor,
      getTableColumnType,
      getTableColumnIconType,
      disabledCategories,
      setPages: setPages || setLocalPages,
      pages: pages.length > 0 ? pages : localPages,
      pagesPerFolder,
      folderLimit,
      sortCategoryAlphabeticallyAndControlLimits
    }
  } else {
    tableProps = {
      getTableColumnColor,
      getTableColumnType,
      getTableColumnIconType,
      setPages: setLocalPages,
      pages: localPages,
      pagesPerFolder,
      folderLimit,
      tableHeaderData,
      sortCategoryAlphabeticallyAndControlLimits
    }
  }

  return (
    <StyledTable>
      {
        rows.length === 0
          ? (
            <div className='empty-row'>
              <h4>There is no data to display.</h4>
            </div>
          )
          : type === 'draggable' ? <DraggableTable {...tableProps as DraggableTableProps} /> : <SortableTable {...tableProps as SortableTableProps} />
      }
    </StyledTable>
  )
}

const StyledTable = styled.div`
  background-color: var(--color-darker);
  width: 100%;
  height: 100%;
  border-radius: 10px;
  overflow-y: auto;

  .empty-row {
    display: flex;
    flex-direction: row;
    justify-content: center;

    h4 {
      color: var(--color-white)
    }
  }

  table {
    padding: 0 .7rem;
    width: 100%;
    border-collapse: separate;
    border-spacing: 0 2px;

    th,
    td {
      margin: 0;
      border: 0;
      padding: 0.6rem 0;
      text-align: left;
      text-transform: capitalize;
      color: var(--color-white);
      font-size: var(--text-md);
      filter: brightness(85%);
    }

    thead{
      &:before {
        display:block;
        text-indent:-99999px;
        line-height:1rem;
        content:"@";
      }

      tr{
        th {
          background-color: var(--color-darker);

          &:first-child {
            padding-left: 1rem;
            border-top-left-radius: 5px;
            border-bottom-left-radius: 5px;
          }

          &.sortable {
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            align-items: center;
            margin-bottom: 0;
            padding-right: 1rem;

            .inverted {
              display: flex;
              transform: rotate(-180deg);
            }
            svg {
              background-color: var(--color-darker);
              padding: 5px;
              width: 20px;
              height: 20px;
              filter: brightness(70%);
              border-radius: 5px;
            }
          }
          &:last-child{
            border-top-right-radius: 5px;
            border-bottom-right-radius: 5px;
          }
        }
      }
    }

    tbody{
      border-radius: 5px;

      &:before {
        display:block;
        text-indent:-99999px;
        line-height:1rem;
        content:"@";
      }

      &:last-child {
        &:after {
          display:block;
          text-indent:-99999px;
          line-height:1rem;
          content:"@";
        }
      }

      tr{
        border-radius: 5px;

        td {
          border-top: 0;
          border-bottom: 1px solid rgba(0,0,0,0.5);
          background-color: var(--color-dark);
          padding: .6rem 0;

          &:first-child{
            padding-left: 1rem;
            border-top-left-radius: 5px;
            border-bottom-left-radius: 5px;
          }
          &:last-child{
            border-top-right-radius: 5px;
            border-bottom-right-radius: 5px;
          }

          &.category-name {
            display: flex;
            flex-direction: row;
            align-items: center;

            span {
              margin-left: .7rem;
            }
          }

          span.blue, span.black {
            background-color: rgb(59, 185, 255);
            padding: 2px 5px;
            font-size: var(--text-md);
            font-weight: var(--font-weight-bd);
            border-radius: 5px;

            svg {
              margin-right: 5px;
            }
          }
          span.black {
            background-color: var(--color-darker);
          }

          div.icon-wrapper{
            display: flex;
            flex-direction: row;

            span {
              margin-right: 5px;
              width: 20px;
              height: 20px;
              
              svg {
                background-color: var(--color-darker);
                padding: 5px;
                width: 20px;
                height: 20px;
                border-radius: 5px;
              }
            }

            img.icon {
              display: block;
              margin-right: 2px;
              width: 20px;
              height: 20px;
              border-radius: 50%;
            }
          }
        }

        &.draggable{
          &:hover {
            cursor: pointer;
          }
            
          td:first-child{
            &:before{
              display: flex;
              position: absolute;
              left: -2px;
              background-color: var(--color-white-fix);
              filter: brightness(70%);
              width: 4px;
              height: 20px;
              content:"";
              border-radius: 2px;
            }
          }
        }

        &:not(:first-child).divider {
          &:before {
            display:block;
            text-indent:-99999px;
            line-height:1rem;
            content:"@";
          }
        } 
      }
    }
  }`

export default Table
