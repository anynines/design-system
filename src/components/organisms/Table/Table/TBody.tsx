import React from 'react'
import styled from 'styled-components'

// I N T E RF A C E S
export interface TBodyProps {
  className?: string
}

// C O M P O N E N T
export const TBody: React.FC<TBodyProps> = (props) => {
  const { children, className } = props

  return (
    <StyledTBody className={className}>
      {children}
    </StyledTBody>
  )
}

// S T Y L E S
const StyledTBody = styled.tbody`
  tr:last-child td {
    border-bottom: 3px solid var(--color-black);
  }

  &:last-child{
    tr {
      &:last-child {
        td:first-child {
          border-bottom-left-radius: var(--radius);
        }
        td:last-child {
          border-bottom-right-radius: var(--radius);
        }
      } 
    }
  }
`
