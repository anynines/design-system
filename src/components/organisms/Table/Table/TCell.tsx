import React from 'react'
import styled from 'styled-components'

// I N T E R F A C E S
export interface TCellProps {
  value?: number | string
  className?: string
}

// C O M P O N E N T
export const TCell: React.FC<TCellProps> = (props) => {
  const { value, children, className } = props

  return (
    <StyledTCell className={className}>
      {value || children}
    </StyledTCell>
  )
}

// S T Y L E S
const StyledTCell = styled.td`
  border: none;
  background-color: var(--color-dark-50);
  padding: 10px;
  text-align: left;
  white-space: nowrap;
  font-size: 14px;

  &:first-child {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--color-dark);
    text-align: left;

    a {
      display: flex;
      flex-direction: center;
      justify-content: center;
    }
  }

  &:last-child {
    background-color: var(--color-dark);
  }

  &.highlight-primary {
    border-left: 5px solid var(--color-primary);
  }

  &.green {
    background-color: var(--color-success) !important;
  }

  &.red {
    background-color: var(--color-error) !important;
  }
`
