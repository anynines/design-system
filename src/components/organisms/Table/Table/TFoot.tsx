import React from 'react'
import styled from 'styled-components'

// I N T E R F A C E S
export interface TFootProps {
  className?: string
}

// C O M P O N E N T
export const TFoot: React.FC<TFootProps> = (props) => {
  const { children, className } = props

  return (
    <StyledTFoot className={className}>
      {children}
    </StyledTFoot>
  )
}

// S T Y L E S
const StyledTFoot = styled.tfoot`
  td {
    font-weight: 900;
  }
  
  &:last-child{
    tr {
      &:last-child {
        td:first-child {
          border-bottom-left-radius: 5px;
        }
        td:last-child {
          border-bottom-right-radius: 5px;
        }
      } 
    }
  }
`
