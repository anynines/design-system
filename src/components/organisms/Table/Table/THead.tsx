import React from 'react'
import styled from 'styled-components'

// I N T E R F A C E S
export interface THeadProps {
  className?: string
}

// C O M P O N E N T
export const THead: React.FC<THeadProps> = (props) => {
  const { children, className, ...rest } = props

  return (
    <StyledTHead className={className} {...rest}>
      {children}
    </StyledTHead>
  )
}

// S T Y L E S
const StyledTHead = styled.thead`
  tr {
    &:last-child {
      td:first-child {
        border-top-left-radius: var(--radius);
      }
      td:last-child {
        border-top-right-radius: var(--radius);
      }
    } 
  }
`
