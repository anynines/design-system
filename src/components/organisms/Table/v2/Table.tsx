import React, { Dispatch, SetStateAction } from 'react'
import styled from 'styled-components'

import { getChildrenWithNewProps } from '../../../../helpers'

// T Y P E S
export interface TableRow {
  [key: string]: string | boolean | number | string[] | number[]
}

export type TableSortDirection = 'asc' | 'desc' | null

export interface TableProps {
  data: TableRow[]
  sortDirection?: TableSortDirection
  sortedBy?: string
}

export interface TableChildrenProps {
  data: TableRow[]
  sortDirection: TableSortDirection
  sortedBy: string
  setSortDirection: Dispatch<SetStateAction<TableSortDirection>>
  setData: Dispatch<SetStateAction<TableRow[]>>
  setSortedBy: Dispatch<SetStateAction<string>>
  onSort: (field: string) => void
}

// H E L P E R S
export const getCellValue = (row: TableRow, field: string): boolean | number | string => {
  const value = row[field]
  if (Array.isArray(value)) return ''
  if (Number.isInteger(value)) return value
  if (typeof value === 'string') return value.toLowerCase()
  return ''
}

// C O M P O N E N T S
export const Table: React.FC<TableProps> = (props) => {
  const {
    data,
    sortDirection,
    sortedBy,
    children
  } = props

  const [currentData, setCurrentData] = React.useState<TableRow[]>([] as TableRow[])
  const [currentSortedBy, setCurrentSortedBy] = React.useState<string>('')
  const [currentSortDirection, setCurrentSortDirection] = React.useState<TableSortDirection>(null)

  React.useEffect(() => {
    if (data.length) {
      setCurrentData(data)
      if (sortedBy) setCurrentSortedBy(sortedBy)
      if (sortDirection) setCurrentSortDirection(sortDirection)
    }
  }, [data, sortDirection, sortedBy])

  const getDirection = (direction: TableSortDirection): TableSortDirection => {
    if (direction === 'asc') return 'desc'
    if (direction === 'desc') return null
    return 'asc'
  }

  const handleOnSort = (field: string): void => {
    const newData = [...currentData]

    if (field !== currentSortedBy) {
      setCurrentSortedBy(field)
      setCurrentSortDirection('asc')
    } else setCurrentSortDirection(getDirection(currentSortDirection))

    if (newData.length) {
      const value = newData[0][field]
      if (Array.isArray(value) || typeof value === 'boolean') return
      if (sortDirection === null) {
        setCurrentData(data)
        return
      }
    }

    const sortedData = newData.sort((firstItem, secondItem) => {
      const firstValue: string | boolean | number = getCellValue(firstItem, field)
      const secondValue: string | boolean | number = getCellValue(secondItem, field)

      if (typeof firstValue === 'string' && typeof secondValue === 'string') return firstValue.charCodeAt(0) - secondValue.charCodeAt(0)
      if (typeof firstValue === 'number' && typeof secondValue === 'number') return Number(firstValue) - Number(secondValue)
      return 0
    })
    if (sortDirection === 'desc') sortedData.reverse()

    setCurrentData(sortedData)
  }

  const childrenProps: TableChildrenProps = {
    data: currentData,
    sortDirection: currentSortDirection,
    sortedBy: currentSortedBy,
    setSortDirection: setCurrentSortDirection,
    setData: setCurrentData,
    setSortedBy: setCurrentSortedBy,
    onSort: handleOnSort
  }

  return (
    <StyledTable>
      {getChildrenWithNewProps(children, childrenProps)}
    </StyledTable>
  )
}

const StyledTable = styled.table`
`
