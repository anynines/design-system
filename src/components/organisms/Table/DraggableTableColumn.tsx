import React  from 'react'
import { Cell } from 'react-table'

import { TableRow, TableAccessor, TableColumnCell, TableColumnIcon, TableColumnCellColor } from './Table'
import TableCell from './TableCell'

interface DraggableTableColumnProps {
  cell: Cell<TableRow, any> // eslint-disable-line @typescript-eslint/no-explicit-any
  cellIndex: number
  getTableColumnColor: (type: TableAccessor | null) => TableColumnCellColor | null
  getTableColumnType: (type: TableAccessor | null) => TableColumnCell | null
  getTableColumnIconType: (type: TableAccessor | null) => TableColumnIcon | null
  isLastColumn: boolean
}

const DraggableTableColumn: React.FC<DraggableTableColumnProps> = (props) => {
  const {
    cell, getTableColumnColor, getTableColumnType, getTableColumnIconType, isLastColumn = false
  } = props

  let header: TableAccessor | null = cell.column.Header as TableAccessor
  if (header === undefined || Object.keys(header).length === 0) header = null

  const className: TableColumnCellColor | null = getTableColumnColor(header)
  const cellType: TableColumnCell | null = getTableColumnType(header)
  const iconType: TableColumnIcon | null = getTableColumnIconType(header)

  return (
    <td role='cell' className={isLastColumn ? 'table-row-link' : ''}>
      <TableCell className={className} cellType={cellType} cell={cell} iconType={iconType} />
      {isLastColumn && <TableCell cellType='link' cell={cell} />}
    </td>
  )
}

export default DraggableTableColumn
