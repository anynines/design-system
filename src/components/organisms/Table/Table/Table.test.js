import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import { Table } from './Table'
import { TRow } from './TRow'
import { THead } from './THead'
import { TBody } from './TBody'
import { THeadCell } from './THeadCell'
import { TCell } from './TCell'

describe('Basic Table', () => {
  let component
  let tHeadData
  let tBodyData

  beforeEach(() => {
    tHeadData = ['ID', 'Name', 'Age', 'Gender']
    tBodyData = [
      {
        id: 1,
        employee: 'Eileen Sherrill',
        age: 30,
        gender: 'female'
      },
      {
        id: 2,
        employee: 'Deborah Kelley',
        age: 20,
        gender: 'female'
      },
      {
        id: 3,
        employee: 'Kay Rich',
        age: 50,
        gender: 'male'
      },
      {
        id: 4,
        employee: 'Frances Douglas',
        age: 40,
        gender: 'male'
      }
    ]
    component = render(
      <Table data-testid='basic-table'>
        <THead data-testid='basic-table-head'>
          <TRow>
            {tHeadData.map((colName, index) => {
              return <THeadCell key={index.toString() + colName}>{colName}</THeadCell>
            })}
          </TRow>
        </THead>
        <TBody data-testid='basic-table-body'>
          {tBodyData.map((row) => {
            return (
              <TRow key={row.id.toString()}>
                <TCell>{row.id}</TCell>
                <TCell>{row.employee}</TCell>
                <TCell>{row.age}</TCell>
                <TCell>{row.gender}</TCell>
              </TRow>
            )
          })}
        </TBody>
      </Table>
    )
  })

  describe('snapshot', () => {
    it('should match default snapshot', () => {
      const { asFragment } = component

      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('rendering', () => {
    describe('table', () => {
      it('should have only 1 table element', () => {
        const { queryAllByTestId } = component
        const table = queryAllByTestId('basic-table')

        expect(table).toHaveLength(1)
      })
    })

    describe('tHead', () => {
      it('should have only 1 thead element', () => {
        const { queryAllByTestId } = component
        const tHead = queryAllByTestId('basic-table-head')

        expect(tHead).toHaveLength(1)
      })

      it('should have correct number of th elements', () => {
        const { container } = component
        const headCols = container.querySelectorAll('th')

        expect(headCols).toHaveLength(tBodyData.length)
      })

      it('should have correct names on th elements', () => {
        const { container } = component
        const headCols = container.querySelectorAll('th')

        headCols.forEach((th, index) => {
          expect(th.innerHTML).toEqual(tHeadData[index])
        })
      })
    })

    describe('tBody', () => {
      it('should have only 1 tBody element', () => {
        const { queryAllByTestId } = component
        const tBody = queryAllByTestId('basic-table-body')

        expect(tBody).toHaveLength(1)
      })

      it('should have correct number of tr elements', () => {
        const { queryByTestId } = component
        const tBody = queryByTestId('basic-table-body')
        const bodyRows = tBody.querySelectorAll('tr')

        expect(bodyRows).toHaveLength(tBodyData.length)
      })

      it('should have correct number of td elements', () => {
        const { queryByTestId } = component
        const body = queryByTestId('basic-table-body')
        const rows = body.querySelectorAll('tr')

        rows.forEach((tr) => {
          const cells = tr.querySelectorAll('td')

          expect(cells).toHaveLength(tHeadData.length)
        })
      })

      it('should have correct values on td elements', () => {
        const { queryByTestId } = component
        const body = queryByTestId('basic-table-body')
        const rows = body.querySelectorAll('tr')

        rows.forEach((tr, trIndex) => {
          const cells = tr.querySelectorAll('td')

          expect(cells[0].innerHTML).toEqual(tBodyData[trIndex].id.toString())
          expect(cells[1].innerHTML).toEqual(tBodyData[trIndex].employee)
          expect(cells[2].innerHTML).toEqual(tBodyData[trIndex].age.toString())
          expect(cells[3].innerHTML).toEqual(tBodyData[trIndex].gender)
        })
      })
    })

    // // Loop through each row and check the content
    // rows.forEach((tr, rowIndex) => {
    //   const cells = tr.find('td')
    //   expect(cells).toHaveLength(cols.length)
    //   expect(cells.at(0).text()).toEqual(data[rowIndex].id)
    //   expect(cells.at(1).text()).toEqual(data[rowIndex].name)
    //   expect(cells.at(2).text()).toEqual(data[rowIndex].email)
    // })
  })
})
