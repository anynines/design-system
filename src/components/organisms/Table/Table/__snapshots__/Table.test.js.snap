// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Table snapshot should match default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  overflow: hidden;
  width: 100%;
  border-radius: calc(var(--radius) * 2);
}

.c0.scrollable {
  display: block;
  overflow: auto;
}

.c5 {
  border: none;
  background-color: var(--color-black-50);
  padding: 10px;
  text-align: left;
  white-space: nowrap;
  font-size: 14px;
}

.c5:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
}

.c5:first-child a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: center;
  -ms-flex-direction: center;
  flex-direction: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5.highlight-primary {
  border-left: 5px solid var(--color-primary);
  background-color: var(--color-dark-80);
}

.c5.green {
  background-color: var(--color-success) !important;
}

.c5.red {
  background-color: var(--color-error) !important;
}

.c2 td {
  border-left: 5px solid transparent;
  color: var(--color-white-70);
  -webkit-transition: var(--transition);
  transition: var(--transition);
}

.c2 td:first-child,
.c2 td:last-child {
  color: var(--color-white);
}

.c2:hover td {
  background-color: var(--color-primary);
  color: var(--color-white);
}

.c2.opened td {
  background-color: var(--color-dark);
}

.c2.clickable {
  cursor: pointer;
}

.c2.green td {
  background-color: var(--color-success-dark);
}

.c2.red td {
  background-color: var(--color-error-dark);
}

.c2.green td:first-child,
.c2.green td:last-child {
  background-color: var(--color-success);
}

.c2.red td:first-child,
.c2.red td:last-child {
  background-color: var(--color-error);
}

.c1 tr:last-child td:first-child {
  border-top-left-radius: var(--radius);
}

.c1 tr:last-child td:last-child {
  border-top-right-radius: var(--radius);
}

.c4 tr:last-child td {
  border-bottom: 3px solid var(--color-black);
}

.c4:last-child tr:last-child td:first-child {
  border-bottom-left-radius: var(--radius);
}

.c4:last-child tr:last-child td:last-child {
  border-bottom-right-radius: var(--radius);
}

.c3 {
  border: none;
  background-color: var(--color-black);
  padding: 10px;
  text-align: left;
  font-size: 14px;
}

<table
    class="c0"
    data-testid="basic-table"
  >
    <thead
      class="c1"
      data-testid="basic-table-head"
    >
      <tr
        class="c2"
      >
        <th
          class="c3"
        >
          ID
        </th>
        <th
          class="c3"
        >
          Name
        </th>
        <th
          class="c3"
        >
          Age
        </th>
        <th
          class="c3"
        >
          Gender
        </th>
      </tr>
    </thead>
    <tbody
      class="c4"
      data-testid="basic-table-body"
    >
      <tr
        class="c2"
      >
        <td
          class="c5"
        >
          1
        </td>
        <td
          class="c5"
        >
          Eileen Sherrill
        </td>
        <td
          class="c5"
        >
          30
        </td>
        <td
          class="c5"
        >
          female
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c5"
        >
          2
        </td>
        <td
          class="c5"
        >
          Deborah Kelley
        </td>
        <td
          class="c5"
        >
          20
        </td>
        <td
          class="c5"
        >
          female
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c5"
        >
          3
        </td>
        <td
          class="c5"
        >
          Kay Rich
        </td>
        <td
          class="c5"
        >
          50
        </td>
        <td
          class="c5"
        >
          male
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c5"
        >
          4
        </td>
        <td
          class="c5"
        >
          Frances Douglas
        </td>
        <td
          class="c5"
        >
          40
        </td>
        <td
          class="c5"
        >
          male
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
